web/src/settings_config.ts

```diff
@@ -702,6 +703,55 @@ export const group_setting_labels = {
     },
 };
 
+// Order of subsections and its settings is important here as
+// this object is used for rendering the assigned permissions
+// in group permissions panel.
+export const realm_group_permission_settings: {
+    subsection_heading: string;
+    settings: RealmGroupSettingName[];
+}[] = [
+    {
+        subsection_heading: $t({defaultMessage: "Joining the organization"}),
+        settings: ["can_invite_users_group", "create_multiuse_invite_group"],
+    },
+    {
+        subsection_heading: $t({defaultMessage: "Channel permissions"}),
+        settings: [
+            "can_create_public_channel_group",
+            "can_create_web_public_channel_group",
+            "can_create_private_channel_group",
+            "can_add_subscribers_group",
+        ],
+    },
+    {
+        subsection_heading: $t({defaultMessage: "Group permissions"}),
+        settings: ["can_manage_all_groups", "can_create_groups"],
+    },
+    {
+        subsection_heading: $t({defaultMessage: "Direct message permissions"}),
+        settings: ["direct_message_permission_group", "direct_message_initiator_group"],
+    },
+    {
+        subsection_heading: $t({defaultMessage: "Moving messages"}),
+        settings: [
+            "can_move_messages_between_topics_group",
+            "can_move_messages_between_channels_group",
+        ],
+    },
+    {
+        subsection_heading: $t({defaultMessage: "Message deletion"}),
+        settings: ["can_delete_any_message_group", "can_delete_own_message_group"],
+    },
+    {
+        subsection_heading: $t({defaultMessage: "Guests"}),
+        settings: ["can_access_all_users_group"],
+    },
+    {
+        subsection_heading: $t({defaultMessage: "Other permissions"}),
+        settings: ["can_add_custom_emoji_group"],
+    },
+];
```

Comment:
Will we get an obvious failure/ exception if we forget to update this when adding a new realm setting?

Just thinking if we rebase some random PR like the new resolved-topics permission, it'd be nice if that didn't potentially break this logic.

Reply:
This was the thing I wanted to discuss. I guess currently, there will be a failure when sending a live update event for the setting which is not included in this data structure, but should work fine otherwise.

Would have been better to directly use settings from `realm.server_supported_permission_settings` field, but as we needed to divide fields in different subsections and then also had to have them in a particular order, needed a different data structure to handle it. Also, currently in the setting templates all group settings are not arranged continously, so makes it difficult to use a data structure like this.

Reply:
Maybe we can add an on-startup assert that this contains everything in `realm.server_supported_permission_settings`? That would cover the most likely mistakes.